eng/pipelines/templates/stages/publish.yml

```diff
@@ -0,0 +1,579 @@
+stages:
+  - ${{ if eq(variables['Build.Repository.Name'], 'Azure/azure-dev') }}:
+    - stage: PublishCLI
+      dependsOn: Sign
+      condition: >-
+        and(
+          succeeded(),
+          ne(variables['Skip.Release'], 'true'),
+          or(
+            eq('Manual', variables['BuildReasonOverride']),
+            and(
+              eq('', variables['BuildReasonOverride']),
+              eq('Manual', variables['Build.Reason'])
+            )
+          )
+        )
+
+      variables: 
+        - template: /eng/pipelines/templates/variables/image.yml
+        - template: /eng/pipelines/templates/variables/globals.yml
+
+      jobs:
+        - deployment: Publish_Release
+          condition: >-
+            and(
+              succeeded(),
+              ne('true', variables['Skip.Publish'])
+            )
+          environment: azure-dev
+
+          pool:
+            name: $(LINUXPOOL)
```

Comment:
Does this work? I could not get pool vars working in deployment contexts.

Reply:
Probably not. Exercising releases requires a lot of disabling and commenting and isn't always a proper test. I'll update and give it a cursory run.